Sources/SwiftDocC/Infrastructure/Link Resolution/LinkResolver.swift

```diff
@@ -125,6 +122,15 @@ public class LinkResolver {
             fatalError("Only SymbolPathTree.Error errors are raised from the symbol link resolution code above.")
         }
     }
+    
+    /// The context resolves links concurrently for different pages.
+    ///
+    /// Since the link resolver may both read and write to the context to cache externally resolved references, it needs to synchronize those accesses to avoid data races.
+    ///
+    /// > Note:
+    /// > We don't generally want to require a lock around these properties because the context will also render pages in parallel and during rendering the external content
```

Comment:
Oh interesting - is there a more succinct or automatic way of synchronizing attributes that you are avoiding here?

And were we seeing bugs in practice related to synchronizing these dictionaries?

Reply:
We have a `Synchronized` wrapper that enforces synchronization on every access to the wrapped value. 

Yes, I did encounter some data races in the cache of external content when I locally tested this yesterday.